src/main/xerus/monstercat/downloader/SongView.kt

```diff
@@ -101,7 +92,60 @@ class SongView(private val sorter: ObservableValue<ReleaseSorting>) :
 			arrayOf(MenuItem("Expand all") { expandAll() },
 				MenuItem("Collapse all") { expandAll(false) })
 		}
-		contextMenu = ContextMenu(*defaultItems())
+		val item1 = MenuItem("Play") {
+			val selected = selectionModel.selectedItem ?: return@MenuItem
+			GlobalScope.launch {
+				Playlist.clear()
+				val value = selected.value
+				when(value) {
+					is Release -> Player.play(value)
+					is Track -> Player.playTrack(value)
+				}
+			}
+		}
+		val item2 = MenuItem("Add to playlist") {
+			val selected = selectionModel.selectedItem ?: return@MenuItem
+			GlobalScope.launch {
+				val value = selected.value
+				when(value) {
+					is Release -> {
+						value.tracks.forEach { track ->
+							Playlist.add(track)
+						}
+					}
+					is Track -> Playlist.add(value)
+				}
+			}
+		}
+		val item3 = MenuItem("Play next") {
+			val selected = selectionModel.selectedItem ?: return@MenuItem
+			GlobalScope.launch {
+				val value = selected.value
+				when(value) {
+					is Release -> {
+						value.tracks.asReversed().forEach { track ->
+							Playlist.addNext(track)
+						}
+					}
+					is Track -> Playlist.addNext(value)
+				}
+			}
+		}
+		val playlistItems = {
+			arrayOf(item1, item2, item3)
```

Comment:
either name the items or simply inline their creation please

Reply:
Fixed in c64be580d1619f73ee1323adfb9f8687567641f8

Reply:
No, that was not what I meant with inlining. I meant that you could consider creating all the items inside the array creation. But renaming them seems to be the better choice anyways, so stay with that. Still, why are you not creating a static array, as in`val playlistItems = arrayOf(item1, item2, item3)`?

Reply:
Keeping it like this now, as it is not much clearer using an array.